packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/api/IObjectDuplicator.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.views.explorer.services.api;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.representations.IStatus;
+
+/**
+ * Used to provide customized behavior for object duplication.
+ *
+ * @author pcdavid
+ */
+public interface IObjectDuplicator {
+    /**
+     * Wraps configuration settings for object duplication in a single object.
+     */
+    record DuplicationSettings(boolean duplicateContent, boolean copyOutgoingReferences, boolean updateIncomingReferences) {
```

Comment:
This record should probably be extracted in a dedicated file, what do you think about?